src/Message/AbstractRequest.php

```diff
@@ -30,26 +28,15 @@ public function getTransactionId()
 
     protected function sendRequest($method, $endpoint, $data = null)
     {
-        $this->httpClient->getEventDispatcher()->addListener('request.error', function (Event $event) {
-            /**
-             * @var \Guzzle\Http\Message\Response $response
-             */
-            $response = $event['response'];
-
-            if ($response->isError()) {
-                $event->stopPropagation();
-            }
-        });
-
         $httpRequest = $this->httpClient->createRequest(
             $method,
             $this->endpoint . $endpoint,
             array(
                 'Authorization' => 'Bearer ' . $this->getApiKey()
             ),
-            $data
+            json_encode($data)
```

Comment:
Yeah I understand, but I think the old behaviour was to use `http_build_query($data)`, instead of `json_encode($data)`. See https://github.com/guzzle/guzzle3/blob/master/src/Guzzle/Http/EntityBody.php#L47
I added the old behaviour here: https://github.com/thephpleague/omnipay-common/pull/147/files#diff-0fc5e771700e739aaf73f2e1d3e66237R42
But that would always use http_build_query.

So either the old was was wrong, the new one is wrong, or the API accepts both. (Or I'm missing an automatic `json_encode` somewhere...)